### DISCUSSION TITLE
Querying message delivery information via API

### DISCUSSION THREAD

**Discussion by abhas** [postalserver/postal#1450] [Help with using Postal]
I need to fetch the list of email submitted for delivery over SMTP and then their corresponding status so that a report can be generated. Is there a way to run queries via the API so that I can use filters similar to the web interface and get these details programmatically? I would be glad to write this utility (in Ruby) and release it so that it might be useful to others as well.

**Comment by willpower232** [comment]
It doesn't look like there is any method for this currently, you could have a look at the existing API controllers and make a new one if you wanted.

https://github.com/postalhq/postal/blob/a735219783127cd0495246ccee89422069da1a1c/api/controllers/messages_api_controller.rb
